glib-macros/tests/test.rs

```diff
@@ -280,3 +280,85 @@ fn subclassable() {
         }
     }
 }
+
+#[test]
+fn gclosure() {
+    let empty = glib::gclosure!(|| {});
+    assert!(empty.invoke(&[]).is_none());
+
+    let no_arg = glib::gclosure!(|| 2i32);
+    assert_eq!(no_arg.invoke(&[]).unwrap().get::<i32>().unwrap(), 2);
+
+    let add_1 = glib::gclosure!(|x: i32| x + 1);
+    assert_eq!(add_1.invoke(&[&3i32]).unwrap().get::<i32>().unwrap(), 4);
+
+    let concat_str = glib::gclosure!(|s: &str| s.to_owned() + " World");
+    assert_eq!(
+        concat_str.invoke(&[&"Hello"]).unwrap().get::<String>().unwrap(),
+        "Hello World"
+    );
+
+    let weak_test = {
+        let obj = glib::Object::new::<glib::Object>(&[]).unwrap();
+
+        assert_eq!(obj.ref_count(), 1);
+        let weak_test = glib::gclosure!(@weak obj => move || obj.ref_count());
+        assert_eq!(obj.ref_count(), 1);
+        assert_eq!(weak_test.invoke(&[]).unwrap().get::<u32>().unwrap(), 3);
+        assert_eq!(obj.ref_count(), 1);
+
+        weak_test
+    };
+    assert!(weak_test.invoke(&[]).is_none());
+
+    let strong_test = {
+        let obj = glib::Object::new::<glib::Object>(&[]).unwrap();
+
+        let strong_test = glib::gclosure!(@strong obj => move || obj.ref_count());
+        assert_eq!(strong_test.invoke(&[]).unwrap().get::<u32>().unwrap(), 2);
+
+        strong_test
+    };
+    assert_eq!(strong_test.invoke(&[]).unwrap().get::<u32>().unwrap(), 1);
+
+    let weak_none_test = {
+        let obj = glib::Object::new::<glib::Object>(&[]).unwrap();
+
+        let weak_none_test = glib::gclosure!(@weak-allow-none obj => move || {
+            obj.map(|o| o.ref_count()).unwrap_or_default()
+        });
+        assert_eq!(weak_none_test.invoke(&[]).unwrap().get::<u32>().unwrap(), 2);
+
+        weak_none_test
+    };
+    assert_eq!(weak_none_test.invoke(&[]).unwrap().get::<u32>().unwrap(), 0);
+
+    {
+        let obj1 = glib::Object::new::<glib::Object>(&[]).unwrap();
+        let obj2 = glib::Object::new::<glib::Object>(&[]).unwrap();
+
+        let obj_arg_test = glib::gclosure!(|a: glib::Object, b: glib::Object| {
```

Comment:
This wants a run through `cargo fmt`

Reply:
I ran it, this line is 99 characters long, one short of getting reformatted...